android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt

```diff
@@ -365,25 +367,25 @@ constructor(
           containedResources.forEach { resource ->
             when (resource.resourceType) {
               ResourceType.CareTeam -> {
-                careTeams.add(resource as CareTeam)
+                careTeams.add(updateResourceIds(resource) as CareTeam)
```

Comment:
Since `updateResourceIds(resource)` is called for every resource type (in the when statement) and isn't dependent on the type, would it be better to call that function once before the `when` statement?